app/models/game.rb

```diff
@@ -1,4 +1,34 @@
 class Game < ApplicationRecord
+  include PgSearch
+  pg_search_scope(
+    :search_by, lambda do |type, query|
+      if type == 'main'
+        {
+        :against =>{ title: 'A', description: 'B'},
+        using: {
+          tsearch:{dictionary: "english",
+                   prefix: true,
+                   any_word: true}
+        },
+        :query => query
+      }
+      elsif type == 'user'
+        {
+        :associated_against => {:user => :company},
+        using: {
+          tsearch: {prefix: true,
+                    any_word: true}
+        },
+        :query => query
+      }
+      else {
+        associated_against: {:tags => :id},
+        :query => query
+        }
+      end
```

Comment:
Indenting in the entire `pg_search_scope` is pretty odd.

Also, you are making copious use of the old notation for symbol key-value pairs. (e.g. `:user => :company` should be `user: :company`, `:query => query` should be `query: query`, etc). Please use the new notation whenever possible.